winch/codegen/src/codegen/env.rs

```diff
@@ -386,6 +398,15 @@ impl<'a, 'translation, 'data, P: PtrSize> FuncEnv<'a, 'translation, 'data, P> {
         self.name_intern.clear();
         mem::take(&mut self.name_map)
     }
+
+    /// Lazily calculates and returns [`VMRuntimeLimits`].
+    pub fn resolve_vmruntime_limits(&mut self) -> &VMRuntimeLimits {
+        self.vmruntime_limits
+            .get_or_insert_with(|| VMRuntimeLimits {
+                ptr_offset: self.vmoffsets.ptr.vmctx_runtime_limits() as u32,
+                fuel_consumed_offset: self.vmoffsets.ptr.vmruntime_limits_fuel_consumed() as u32,
```

Comment:
Out of curiosity was the `vmruntime_limits` cache-of-sorts here added because the original methods were too slow? Or because the original `self.vmoffsets.ptr` wasn't accessible when the cache was needed?

Reply:
Not really, I didn't perform extensive benchmarking here, I think it might be fine to immediately resolve the limits when constructing `FuncEnv`.

Reply:
Ah ok makes sense, I ask because it felt a bit unidiomatic to copy a few fields from `VMOffsets` into a separate structure vs threading around the `VMOffsets` themselves, but it's also a pretty minor point